### PR TITLE
Telemetry redis

### PR DESCRIPTION
**Pull request by chillaq**
# Python SDK

## Extra Notes

### CODE REVIEW

File: splitio/storage/redis.py [hunk 1]
```diff
@@ -180,27 +180,22 @@ def get_split_names(self):
...
+        """
+        return 0
+
     def get_all_splits(self):
         """
         Return all the splits in cache.
 
-        :return: List of all splits in cache.
-        :rtype: list(splitio.models.splits.Split)
+        :return: 0
```
Comment (mmelograno) [inline splitio/storage/redis.py:195]: we are using this method for manager, why you removed the logic?

File: splitio/storage/adapters/redis.py [hunk 2]
```diff
@@ -297,6 +312,54 @@ def pipeline(self):
...
+        host_ip = 'Unknown'
+        try:
+            host_ip = _get_ip()
+            host_name = socket.gethostname()
+        except:
+            _LOGGER.debug("Could not get hostname or ip")
+            pass
+        return host_name, host_ip
+
+    def record_stats(self, values):
```
Comment (mmelograno) [inline splitio/storage/adapters/redis.py:333]: Redis won't record stats periodically, you will record latencies and exceptions at the time it happens

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: splitio/recorder/recorder.py [hunk 3]
```diff
@@ -131,22 +136,23 @@ def record_treatment_stats(self, impressions, latency, operation):
...
 
-    def record_track_stats(self, event):
+    def record_track_stats(self, event, latency):
         """
         Record stats for tracking events.
 
         :param event: events tracked
         :type event: splitio.models.events.EventWrapper
         """
+        self._telemetry_redis_storage.record_latency(MethodExceptionsAndLatencies.TRACK.value, latency)
```
Comment (mmelograno) [inline splitio/recorder/recorder.py:157]: this should be also pipelined

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: splitio/recorder/recorder.py (hunk 3 above)
Comment (chillaq) [reply splitio/recorder/recorder.py:157]: Yep, I added the pipe in the RedisTelemetryStorage class

File: splitio/storage/adapters/redis.py [hunk 4]
```diff
@@ -297,6 +312,54 @@ def pipeline(self):
...
             raise RedisAdapterException('Error executing ttl operation') from exc
 
+    def record_init(self, *values):
+        try:
+            host_name, host_ip = self._get_host_info()
+            return self.hset(TELEMETRY_CONFIG_KEY, 'python-' + __version__ + '/' + host_name+ '/' + host_ip, str(*values))
+        except RedisError as exc:
+            raise RedisAdapterException('Error pushing telemetry config operation') from exc
+
+    def _get_host_info(self):
```
[the remark this answers is not in the document]
Comment (chillaq) [reply splitio/storage/adapters/redis.py]: fixed

File: splitio/storage/redis.py (hunk 1 above)
Comment (chillaq) [reply splitio/storage/redis.py]: I added it back

**Review by EmilianoSanchez — APPROVED** [review]
LGTM

**Pull request merged**